contracts/interactiveAdapters/Akropolis/AkroStakingInteractiveAdapter.sol

```diff
@@ -0,0 +1,118 @@
+import { SafeERC20 } from "../../shared/SafeERC20.sol";
+import { TokenAmount } from "../../shared/Structs.sol";
+import { ERC20ProtocolAdapter } from "../../adapters/ERC20ProtocolAdapter.sol";
+import { InteractiveAdapter } from "../InteractiveAdapter.sol";
+
+/**
+ * @dev Staking contract interface.
+ * Only the functions required for AkroStakingAdapter contract are added.
+ */
+interface AkroStaking {
+    function stake(uint256 amout, bytes calldata _data) external;
+
+    function stakeFor(
+        address _user,
+        uint256 _amount,
+        bytes memory _data
+    ) external;
+
+    function unstakeAllUnlocked(bytes calldata _data) external returns (uint256);
+}
+
+/**
+ * @title Interactive Adapter for Akropolis Staking protocol.
+ * @dev Implementation of ProtocolAdapter interface.
+ * @author Alexander Mazaletskiy <am@akropolis.io>
+ */
+
+/**
+ * @title Interactive adapter for Compound protocol.
+ * @dev Implementation of InteractiveAdapter abstract contract.
+ */
+contract AkroStakingInteractiveAdapter is InteractiveAdapter, ERC20ProtocolAdapter {
+    using SafeERC20 for ERC20;
+
+    address internal constant STAKING = 0x3501Ec11d205fa249f2C42f5470e137b529b35D0;
+
+    /**
+     * @notice Stake ADEL token to the ADEL Staking
+     * @param tokenAmounts Array with one element - TokenAmount struct with
+     * ADEL token address, token amount to be stake, and amount type.
+     * @return tokensToBeWithdrawn empty array
+     * @dev Implementation of InteractiveAdapter function.
+     */
+    function deposit(TokenAmount[] calldata tokenAmounts, bytes calldata data)
+        external
+        payable
+        override
+        returns (address[] memory tokensToBeWithdrawn)
+    {
+        require(tokenAmounts.length == 1, "AKROIA: should be 1 tokenAmount[1]");
+
+        address token = tokenAmounts[0].token;
+        uint256 amount = getAbsoluteAmountDeposit(tokenAmounts[0]);
+
+        ERC20(token).safeApprove(STAKING, amount, "AKROIA");
+
+        //get user data from calldata
+        address userAddress = abi.decode(data, (address));
+
+        // solhint-disable-next-line no-empty-blocks
+        try AkroStaking(STAKING).stakeFor(userAddress, amount, "0x")  {} catch Error(
+            string memory reason
+        ) {
+            revert(reason);
+        } catch {
+            revert("AKROIA: stake fail");
+        }
+    }
+
+    /**
+     * @notice Withdraws tokens from the Adel Staking
+     * @param tokenAmounts Array with one element - TokenAmount struct with
+     *  ADEL address, ADEL amount to be unstake
+     * @return tokensToBeWithdrawn Array with one element - ADEL token.
+     * @dev Implementation of InteractiveAdapter function.
+     */
+    function withdraw(TokenAmount[] calldata tokenAmounts, bytes calldata)
+        external
+        payable
+        override
+        returns (address[] memory tokensToBeWithdrawn)
+    {
+        require(tokenAmounts.length == 1, "AKROIA: should be 1 tokenAmount[2]");
```

Comment:
`tokenAmounts` should be empty and `tokensToBeWithdrawn` should be encoded in `bytes data` variable 
decode it like line 79